From 9d4a1ab7d93b7f5b3cfcfb9823d6e53ad2bc8063 Mon Sep 17 00:00:00 2001
From: Alex Marginean <alexandru.marginean@nxp.com>
Date: Mon, 23 Sep 2019 14:23:35 +0300
Subject: [PATCH 672/741] felix: skip probing if device is disabled in DT

commit 3fd48be50104e194bd4a1980ac741eb89fe52971 from
http://source.codeaurora.org/external/qoriq/qoriq-components/linux

For some of the SoC configurations the switch HW is disabled but the PCI
function is still presented on the bus.  To avoid boot time errors the
function must be disabled and the patch adds support for that.

Signed-off-by: Alex Marginean <alexandru.marginean@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 drivers/net/ethernet/mscc/felix_board.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/drivers/net/ethernet/mscc/felix_board.c b/drivers/net/ethernet/mscc/felix_board.c
index 27a3c170a102..04e3025c437f 100644
--- a/drivers/net/ethernet/mscc/felix_board.c
+++ b/drivers/net/ethernet/mscc/felix_board.c
@@ -733,6 +733,11 @@ static int felix_pci_probe(struct pci_dev *pdev, const struct pci_device_id *id)
 	size_t len;
 	int i, err;
 
+	if (pdev->dev.of_node && !of_device_is_available(pdev->dev.of_node)) {
+		dev_info(&pdev->dev, "device is disabled, skipping\n");
+		return -ENODEV;
+	}
+
 	err = pci_enable_device(pdev);
 	if (err) {
 		dev_err(&pdev->dev, "device enable failed\n");
-- 
2.17.1

